### ISSUE TITLE
Function plugins + @import cause error when cache is turned off

### ISSUE THREAD

**Issue by cryovat** [dotless/dotless#445]
Hi,

I ran into a problem when combining a function plugin and import directives. It works fine with caching turned on, but when caching is turned off, the first request will be served correctly, and subsequent requests of the .less file will crash with an exception on the form of:

```
Function 'test_func' already exists in environment but is added by plugin TestPlugin
   at dotless.Core.Parser.Infrastructure.Env.AddPlugin(IPlugin plugin)
   at dotless.Core.LessEngine.TransformToCss(String source, String fileName)
   at dotless.Core.ParameterDecorator.TransformToCss(String source, String fileName)
   at dotless.Core.HandlerImpl.Execute() i c:\projects\dotless\src\dotless.AspNet\HandlerImpl.cs:linje 29
   at dotless.Core.LessCssHttpHandler.ProcessRequest(HttpContext context) i c:\projects\dotless\src\dotless.AspNet\LessCssHttpHandler.cs:linje 18
   at System.Web.HttpApplication.CallHandlerExecutionStep.System.Web.HttpApplication.IExecutionStep.Execute()
   at System.Web.HttpApplication.ExecuteStep(IExecutionStep step, Boolean& completedSynchronously)
```
Verified with dotLess 1.5.0.0 beta 4 in a fresh ASP.NET 4.5 web application with the following source (the plugin was defined within the main web application project itself; not in a library).

##### TestPlugin.cs
```csharp
namespace WebApplication1
{
    public class TestPlugin : IFunctionPlugin
    {
        public Dictionary<string, Type> GetFunctions()
        {
            return new Dictionary<string, Type>()
            {
                {"test_func", typeof (TestDotLessFunction)}
            };
        }
    }

    public class TestDotLessFunction : Function
    {
        protected override Node Evaluate(Env env)
        {
            return new Quoted("hello", true);
        }
    }
}
```

##### Web.config
```xml
  <dotless minifyCss="false" cache="false" web="true" strictMath="false">
    <plugin name="TestPlugin" assembly="WebApplication1" />
  </dotless>
```

##### outer.less
```css
@foo: test_func();

@import "inner.less";
```
##### inner.less
```css
body {
    background-color: pink;
}
```

##### Observed behavior:

First request to outer.less works fine, second causes the handler to crash.

**Comment by rytmis** [comment]
Thanks for reporting. Do you happen to know if this is a regression from 1.4?

**Comment by cryovat** [comment]
Did a test now, and it definitely looks like a regression. Can't reproduce it with 1.4.4 from Nuget.

Also, my bug report is inaccurate. I made it to closely mirror the scenario I encountered the bug in, but the import directive is actually a red herring. I'm able to trigger it with this source file alone:

```css
@foo: test_func();

body {
    background: @foo;
}
```

**Comment by rytmis** [comment]
Thanks again for finding this. I profiled dotless some time back, and I noticed that registering the core functions into evaluation environments was taking a considerable slice of time. I noticed that this was mostly because every time an Env was constructed, we would reflect through the assembly looking for functions. I decided to cache the core functions because they're guaranteed to not change. But instead of adding the cached functions to a new Dictionary for each Env, I boneheadedly just assigned the static, cached Dictionary to the instance field. Best of all was, I distinctly remember _intending_ to do the right thing. :)

Anyway, it should be fixed now. Check out https://www.nuget.org/packages/dotless/1.5.0-beta5
